app/services/api_tracking_information.rb

```diff
@@ -0,0 +1,39 @@
+class ApiTrackingInformation
+  require 'open-uri'
+  require 'pry'
+
+  def self.tracking_by(**args)
+    begin
+      if args[:id]
+        api_url = "https://api.aftership.com/v4/trackings/#{args[:id]}"
+      elsif args[:slug] && args[:tracking_number]
+        api_url = "https://api.aftership.com/v4/trackings/#{args[:slug]}/#{args[:tracking_number]}"
+      end
+
+      serialized = open(
+        api_url,
+        "aftership-api-key" => "711623ab-a860-4d3a-84d1-bfa891ff5058",
+        "Content-Type" => "application/json"
+      ).read
```

Comment:
There're more robust gem handling http request like

https://github.com/jnunemaker/httparty or https://github.com/httprb/http

which handles timeout, errors, connection performance better than just open/close the API connection, you can try few of them and study them : )